simulator/controller/controller.go

```diff
@@ -33,15 +33,48 @@ type controllerInitializersFunc func() (initializers map[string]initFunc)
 
 var _ controllerInitializersFunc = newControllerInitializers
 
-func RunController(client clientset.Interface, c *restclient.Config) (func(), error) {
+type Manager interface {
+	RunController()
+	ShutdownController()
+	ResetController()
+}
+
+type manager struct {
```

Comment:
Added new struct which is named manager, to manage shutdown function and resetting.